src/main/java/com/clevertap/stormdb/Buffer.java

```diff
@@ -202,6 +202,25 @@ int add(int key, byte[] value, int valueOffset) {
         return address;
     }
 
+    /**
+     * Checks the key at the given offset and updates its value with newValue otherwise return false
+     */
+    boolean update(int key, byte[] newValue, int valueOffset, int addressInBuffer) {
+        int savedKey = byteBuffer.getInt(addressInBuffer);
+        if (savedKey != key) {
+            return false;
+        }
+        // get current position in buffer
+        int currentPosition = byteBuffer.position();
```

Comment:
Instead of playing with the position, what do you think of using System.arrayCopy() to copy the value bytes directly into the underlying byte array?

Reply:
For this, I searched around on the web but 95% of the articles used this method only and also using system.arrayCopy will be a little slow because first, we will have to create a new byte buffer and then put the value in it, then copying that will take equivalent to O(valueSize)

Reply:
Oh. I had something like the following in my mind, which doesn't create a new byte array or byte buffer object.
```java
System.arrayCopy(newValue, valueOffset, byteBuffer.array(), addressInBuffer + KEY_SIZE, valueSize);
```

Could that work?

Reply:
let me check this if this works